java/solomiia_tymoshchuk/src/com/zoolatech/lecture1/tasks/_4/CharCounter.java

```diff
@@ -0,0 +1,36 @@
+package com.zoolatech.lecture1.tasks._4;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class CharCounter {
+    private final String input;
+
+    CharCounter(String input) {
+        this.input = input;
+    }
+
+    public Map<String, Long> countCharsInString() {
+        Map<String, Long> map = Arrays.stream(input.split(""))
+                .filter(letter -> !letter.equals(" "))
+                .collect(Collectors.groupingBy(Function.identity(), Collectors.counting()));
+        return map;
+    }
+
+    public LinkedHashMap<String, Integer> sortedMap() {
```

Comment:
the same can be done using the following code:
```
return new TreeMap<>(countCharsInString());
```

Method `Collectors.groupingBy` also accepts a lambda, that can create a TreeMap directly - so, the sorted map can be returned from the method `countCharsInString` with some small modifications.